packages/client/hmi-client/src/services/mit-askem.ts

```diff
@@ -0,0 +1,30 @@
+import API, { Poller } from '@/api/api';
+import { logger } from '@/utils/logger';
+
+export async function codeToAcset(code: string) {
+	const response = await API.post(`code/to_acset`, code);
+	return response.data ?? null;
+}
+
+export async function findVarsFromText(text: string) {
+	const id = await API.post(`/code/annotation/find_text_vars`, text);
+	const poller = new Poller<object>()
+		.setInterval(2000)
+		.setThreshold(90)
+		.setPollAction(async () => {
+			const response = await API.get(`/code/response?id=${id.data}`);
+			if (response) {
+				return {
+					data: response.data.data
+				};
+			}
+			return { error: logger.info(`Linking metadata to model...`) };
```

Comment:
Not sure what you are trying to do here, are you trying to pass the logger into the poller?

Reply:
Just trying to display a status message to indicate that the action is still happening.

Reply:
Should change the message though.